objstore.go

```diff
@@ -284,6 +284,7 @@ type ObjectAttributes struct {
 
 type IterObjectAttributes struct {
 	Name         string
+	Size         int64
```

Comment:
Can the size be negative? Maybe let's use uint64.

Reply:
Maybe as a canary for "this provider does not expose a size in iter"; idk if there is such a provider though